src/main/java/ch/njol/skript/expressions/ExprAdventureRestrictions.java

```diff
@@ -0,0 +1,177 @@
+@Examples({
+	"add dirt to destroyable restrictions of player's tool",
+	"add (stone, diamond ore) to breakable blocks of {_item}",
+	"clear break restrictions of {_item}",
+	"remove sand from destroyable blocks of {_item} in adventure mode"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
+public class ExprAdventureRestrictions extends PropertyExpression<ItemType, ItemType> {
+
+	static {
+		if (Skript.methodExists(ItemMeta.class, "getDestroyableKeys")) {
+			Skript.registerExpression(ExprAdventureRestrictions.class, ItemType.class, ExpressionType.PROPERTY,
+				"[the] (:break|:destroy|:place|:build)able blocks of %itemtypes% [in adventure [mode]]",
+				"[the] (:break|:destroy|:place|:build)[able] restrictions of %itemtypes%",
+				"%itemtypes%'[s] (:break|:destroy|:place|:build)able blocks [in adventure [mode]]",
+				"%itemtypes%'[s] (:break|:destroy|:place|:build)[able] restrictions");
+		}
+	}
+
+	private boolean destroy;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<ItemType>) exprs[0]);
+		this.destroy = parseResult.hasTag("break") || parseResult.hasTag("destroy");
+		return true;
+	}
+
+	@Override
+	protected ItemType @NotNull [] get(Event event, ItemType [] source) {
+		Set<ItemType> existingKeys = new HashSet<>();
+		for (ItemType item : source) {
+			ItemMeta meta = item.getItemMeta();
+			if (destroy ? meta.hasDestroyableKeys() : meta.hasPlaceableKeys()) {
+				for (Namespaced key : destroy ? meta.getDestroyableKeys() : meta.getPlaceableKeys()) {
+					Material material = BukkitUnsafe.getMaterialFromMinecraftId(key.toString());
+					if (material != null) {
+						existingKeys.add(new ItemType(material));
+					}
+				}
+			}
+		}
+		return existingKeys.toArray(new ItemType[0]);
+	}
+
+	@SuppressWarnings("NullableProblems")
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) {
+			return CollectionUtils.array(ItemType[].class);
+		} else if (mode == ChangeMode.RESET || mode == ChangeMode.DELETE) {
+			return CollectionUtils.array();
+		}
+		return null;
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		ItemType[] source = getExpr().getArray(event);
+
+		Set<Namespaced> deltaKeys = new HashSet<>();
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) {
+			for (Object o : delta) {
+				if (o instanceof ItemType) {
+					for (ItemStack stack : ((ItemType) o).getAll())
+						deltaKeys.add(stack.getType().getKey());
+				}
+			}
+		}
+
+		for (ItemType item : source) {
+
+			ItemMeta meta = item.getItemMeta();
+			Collection<Namespaced> newKeys = new ArrayList<>();
+
+			switch (mode) {
+				case RESET:
+				case DELETE:
+                    break;
```

Comment:
indentation?

Reply:
I've got no idea how to fix this, in my ide it looks perfectly normal

Reply:
looks like there are some spaces in the file (in this case, at `break`). double check to make sure that tabs are being used on every line.